spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/dsl/TaskDslParser.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.data.core.dsl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Michael Minella
+ */
+public class TaskDslParser {
+
+	private String expressionString;
+
+	private List<Token> tokenStream;
+
+	private int tokenStreamLength;
+
+	private int tokenStreamPointer; // Current location in the token stream when
+
+	// processing tokens
+
+	private int lastGoodPoint;
+
+	/**
+	 * Parse a task definition without supplying the task name up front.
+	 * The task name may be embedded in the definition.
+	 * For example: <code>myTask = filejdbc</code>
+	 *
+	 * @return the AST for the parsed stream
+	 */
+	public ModuleNode parse(String stream) {
```

Comment:
@markfisher I created [XD-3377](https://jira.spring.io/browse/XD-3377) to handle the refactoring.  One other reason for what I did was the existing code was very stream focused.  A proper refactoring should break it up so common code is stream/task agnostic and then the stream/tasks get specific further in the weeds.